### PR TITLE
Update Main_Processor.md

### PR DESCRIPTION
**Pull request by jmvega52**
Testing Pull Request

### CODE REVIEW

File: Documentation/Signoffs/Main_Processor.md [hunk 1]
```diff
@@ -1,53 +1,80 @@
...
 
-Making tables for constraints and BOM will use Excel to Markdown here:
-https://tabletomarkdown.com/convert-spreadsheet-to-markdown/
+| No. | Specifications and Constraints                     | Origin          |
+| --- | -------------------------------------------------- | --------------- |
+| 1   | Shall be powered from wall outlet                  | Specification   |
+| 2   | Shall process and output information within 1.4 ms | Specification   |
+| 3   | Shall process sounds at a rate of at least 40 kHz  | Specification   |
+| 4   | Shall keep track of outputs reaching 80+ dB        | Greater Impacts |
+| 5   | Shall minimize storage of classroom noise          | Ethics          |
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/Main_Processor.md]: What does this mean specifically?

File: Documentation/Signoffs/Main_Processor.md [hunk 2]
```diff
@@ -1,53 +1,80 @@
...
+| 6   | Shall have a safe power toggle                     | IEC 60950-1     |
+| 7   | Shall not reach 60°C (140°F)                       | Safety          |
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/Main_Processor.md]: How will this be ensured?

Comment (JesseTNRoberts) [reply Documentation/Signoffs/Main_Processor.md]: How can you analyze this?

File: Documentation/Signoffs/Main_Processor.md [hunk 3]
```diff
@@ -1,53 +1,80 @@
...
 
-Example Signoff that was shown in class can be found here:
-https://github.com/Hawk652/Capstone-Guidance-Robot/blob/main/Documentation/Signoffs/Power%20Subsystem.md
+## Buildable schematic 
 
-# Required sections
+![image](https://github.com/CarsonDPope/Active-Noise-Control-With-Wall-Transmission-Detection/blob/jmvega52-patch-1/Documentation/Images/DSPBOARD.png)
+The above image is a schematic of the BF706 processor, the schematic focuses on the most important aspects of this device with respect to the overall design goal. The schematic shows stereo input and output ports along with a micro USB input and the subsystems that will connected to this device.  
+## Analysis
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/Main_Processor.md]: This does not seem to address all the constraints.

Comment (CarsonDPope) [reply Documentation/Signoffs/Main_Processor.md]: Is this in relation to the schematic not addressing the constraints or the paragraph?

File: Documentation/Signoffs/Main_Processor.md (hunk 2 above)
Comment (CarsonDPope) [reply Documentation/Signoffs/Main_Processor.md]: Updated constraint table, will add sections for each constraint

File: Documentation/Signoffs/Error-Signoff.md
Comment (JesseTNRoberts) [inline Documentation/Signoffs/Error-Signoff.md:1]: Why is this here? It should be in a dedicated signoff.

File: Documentation/Signoffs/Main_Processor.md [hunk 4]
```diff
@@ -1,53 +1,79 @@
...
-Example Signoff that was shown in class can be found here:
-https://github.com/Hawk652/Capstone-Guidance-Robot/blob/main/Documentation/Signoffs/Power%20Subsystem.md
+## Constraints
 
-# Required sections
+| No. | Constraints                                                           | Origin            |
+| --- | --------------------------------------------------------------------- | ----------------- |
+| 1   | System shall be powered by standard wall outlets                      | Spec Constraint   |
+| 2   | Shall process and output with a max delay of 1.4 ms*                  | Design Constraint |
+| 3   | Shall have a sample rate of at least 40 kHz                           | Spec Constraint   |
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/Main_Processor.md]: This isn't something that makes sense to spec. What determines the appropriate sample rate of a system?

File: Documentation/Signoffs/Main_Processor.md [hunk 5]
```diff
@@ -1,53 +1,79 @@
...
+| 4   | Shall have at least 16-bit ADC and DAC                                | Spec Constraint   |
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/Main_Processor.md]: Again why is this spec'd? How do we determine appropriate sampling resolution?

File: Documentation/Signoffs/Main_Processor.md [hunk 6]
```diff
@@ -1,53 +1,79 @@
...
+| 5   | Minimum of 2 input signal sources                                     | Design Constraint |
+| 6   | Minimum of 1 output signals                                           | Design Constraint |
+| 7   | Shall receive serial data from a general purpose computer**           | Design Constraint |
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/Main_Processor.md]: Serial is ambiguous.

File: Documentation/Signoffs/Error-Signoff.md
Comment (CarsonDPope) [reply Documentation/Signoffs/Error-Signoff.md:1]: Jared must have accidentally added it to this branch. I'll check.

File: Documentation/Signoffs/Main_Processor.md [hunk 7]
```diff
@@ -1,53 +1,96 @@
...
 
-The actual design artifacts (3d model, Cad files, gerber files, etc) must be uploaded to the documentation folder in the appropriate location.
+![image](https://github.com/CarsonDPope/Active-Noise-Control-With-Wall-Transmission-Detection/blob/jmvega52-patch-1/Documentation/Images/DSPBOARD_REV2.png?raw=true)
+The above image is a schematic of the ADSP-BF706 EZ-KIT Mini, the schematic focuses on the most important aspects of ths device with respect to the overall design goal. The schematic shows stereo input and output ports along with a micro USB input and the subsystems that will be connected to this device.  
 
 ## Analysis
 
-A complete and relevant analysis of the design showing that it **should** meet the constraints and perform the desired function must be given. This analysis must be comprehensive and well explained so that it is convincing to the faculty supervisor. If the signoff request is not convincing either because the requirements and constraints are insufficient, unjustified, or not appropriately shown to be met by the design, then approval will not be given. Without approval, the components for the subsystem aren't allowed to be ordered. 
+#### Input
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/Main_Processor.md:52]: How will you get 2 signals into a single port? And are you sure that the dsp will allow you to handle these separately or are they automatically squashed into a stereo audio codec? 

Need to analyze the constraint regarding input a bit more to show this is met.

File: Documentation/Signoffs/Main_Processor.md [hunk 8]
```diff
@@ -1,53 +1,96 @@
...
+The system will receive two analog signals through the stereo jack<sup>5</sup>. These signals are then sent to the ADAU1761 SigmaDSP low power stereo audio codec which features 24-bit stereo audio ADC and DAC<sup>4</sup>. The rated signal to noise ratio is >98 dB SNR. The system also samples at 48 kHz<sup>3</sup>.The digital samples are then sent to the respective delay line. As each sample is received, the oldest sample in the processor is erased<sup>9</sup>.
+
+#### Power
+The system has to be powered through a USB port connection. This will both power and send the data to the processor during boot. The system can be powered from wall power through a USB charger<sup>1</sup>. USB 2.0 interfaces run off of 5 volts and have a max current draw of 500 mA, for a max power draw of 2.5 Watts. To power the device through the USB connected to the processor, the jumper J2 needs to connect the 5V pin to the 706 pin. Since the board also needs to use the USB port in order to install the designed software, the system will have to be programmed using it's One Time Programmable (OTP) memory as to allow the system to boot without being connected directly to a computer. 
+
+#### Speed and Memory
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/Main_Processor.md:60]: I don't follow your logic regarding the time from input to output. The analysis is too hand wavy. You need to be more precise and thorough in showing that the time is bounded to be less than the constraint.

**Review by JesseTNRoberts — COMMENTED** [review]
If an Arduino is essential to the design it needs to be represented here as attached to the board.

Why are three of these being requested?

**Comment by CarsonDPope** [comment]
*** Question but most frequencies actually decayed before 0.6 seconds or 28800 samples ( and even more rough estimates could be made since the system will adapt anyway allowing smaller filters to work more efficient), which would drastically decrease the processing delay and memory components. We are targeting lower frequency sounds mostly as our target. Refer to reference [8]. Just wanted to know your thoughts on this constraint, it is useful as an overestimation, especially with the simulation itself already being an overestimation. ***

File: Documentation/Signoffs/Main_Processor.md [hunk 9]
```diff
@@ -35,57 +35,57 @@ The goal of this subsystem is to process the input signal, and output an appropr
...
+The filtered-x LMS algorithm requires the current and previous samples for both the input and error signals. These samples are stored in arrays called delay lines. The size of these delay lines are determined by the acoustic impulse response of the room (that is the acoustical response of the room including reflections and diffusion). There are 96000 samples in 2 seconds at 48 kHz. Using 16-bit samples, this is equivalent to 187.5 kiB per delay line. 
 
-$$ 1(2 \times 96000) + 1 = 192001 \text{ MACs } $$
+An estimation of the number of MAC instructions needed per update by the filtered-x LMS algorithm can be found by the equation $IJK(L+M)+K$. I is the number of input channels, J is the number of output channels, and K is the number of error channels. In this case, all of the previous values are equal to 1. L and M are the control filter length and the length of the estimated acoustic impulse response respectively. Using an overestimation of L equal to M, the algorithm will approximately require 192001 MAC innstructions. At 800 millions MACs per second, this is equivalent to 240 microseconds of delay due to math operations.
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/Main_Processor.md:66]: citation needed

**Review by JesseTNRoberts — APPROVED** [review]
Add an inline citation for the estimation of time complexity (I saw the paper in the citation list, but it needs to be cited inline as well). However, I am approving as is, but be sure to make this update.
